clients/src/main/java/org/apache/kafka/common/requests/AbstractResponse.java

```diff
@@ -68,12 +71,15 @@ protected void updateErrorCounts(Map<Errors, Integer> errorCounts, Errors error)
 
     protected abstract Struct toStruct(short version);
 
-    public static AbstractResponse parseResponse(ApiKeys apiKey, Struct struct) {
+    /*
+     * Responses that require a delayed allocation need a reference to their closeable NetworkReceive.
+     */
+    public static AbstractResponse parseResponse(ApiKeys apiKey, Struct struct, Closeable closeable) {
```

Comment:
nit: I'm not sure we need or want the `Closeable` generality here. Could we just use the `NetworkReceive` type?